dkist/wcs/models.py

```diff
@@ -753,27 +763,56 @@ def __init__(self, array_shape, order='C', **kwargs):
         self.order = order
 
     def evaluate(self, *inputs):
-        ravel_shape = self.array_shape[1]
-        if self.order == 'F':
-            inputs = inputs[::-1]
-            ravel_shape = self.array_shape[0]
-        return (np.round(inputs[0]) * ravel_shape) + inputs[1]
+        # If we have units, then all values must have units, hence test for existence only on the first one.
+        # Also, assume all units are the same.
+        if hasattr(inputs[0], "unit"):
+            input_units = inputs[0].unit
+            input_values = [item[0].value for item in inputs]
```

Comment:
In general I always recommend using `to_value` as it's more explicit.
```suggestion
            input_values = [item[0].to_value(u.pix) for item in inputs]
```

Reply:
Done.